src/bankid.ts

```diff
@@ -0,0 +1,315 @@
+import * as path from "path";
+import * as fs from "fs";
+import * as https from "https";
+import axios, { AxiosError, AxiosInstance } from "axios";
+
+//
+// Type definitions for /auth
+//
+
+export interface AuthRequest {
+  endUserIp: string;
+  personalNumber?: string;
+  requirement?: AuthOptionalRequirements;
+}
+
+export interface AuthResponse {
+  autoStartToken: string;
+  orderRef: string;
+}
+
+interface AuthOptionalRequirements {
+  cardReader?: "class1" | "class2";
+  certificatePolicies?: string[];
+  issuerCn?: string[];
+  autoStartTokenRequired?: boolean;
+  allowFingerprint?: boolean;
+}
+
+//
+// Type definitions for /sign
+//
+
+export interface SignRequest extends AuthRequest {
+  userVisibleData: string;
+  userNonVisibleData?: string;
+}
+
+export interface SignResponse extends AuthResponse {}
+
+//
+// Type definitions for /collect
+//
+
+export interface CollectRequest {
+  orderRef: string;
+}
+
+export interface CollectResponse {
+  orderRef: string;
+  status: "pending" | "failed" | "complete";
+  hintCode?: FailedHintCode | PendingHintCode;
+  completionData?: CompletionData;
+}
+
+export interface CompletionData {
+  user: {
+    personalNumber: string;
+    name: string;
+    givenName: string;
+    surname: string;
+  };
+  device: {
+    ipAddress: string;
+  };
+  cert: {
+    notBefore: string;
+    notAfter: string;
+  };
+  signature: string;
+  ocspResponse: string;
+}
+
+export type FailedHintCode =
+  | "expiredTransaction"
+  | "certificateErr"
+  | "userCancel"
+  | "cancelled"
+  | "startFailed";
+
+export type PendingHintCode =
+  | "outstandingTransaction"
+  | "noClient"
+  | "started"
+  | "userSign";
+
+//
+// Type definitions for /cancel
+//
+
+export interface CancelRequest extends CollectRequest {}
+
+export interface CancelResponse {}
+
+//
+// Type definitions for error responses
+//
+
+export interface ErrorResponse {
+  errorCode: BankIdErrorCode;
+  details: string;
+}
+
+export enum BankIdErrorCode {
+  "alreadyInProgress",
+  "invalidParameters",
+  "unauthorized",
+  "notFound",
+  "requestTimeout",
+  "unsupportedMediaType",
+  "internalError",
+  "Maintenance"
+}
+
+export const REQUEST_FAILED_ERROR = "BANKID_NO_RESPONSE";
+
+//
+// Collection of overarching types
+//
+
+export enum BankIdMethod {
+  auth = "auth",
+  sign = "sign",
+  collect = "collect",
+  cancel = "cancel"
+}
+
+export type BankIdRequest =
+  | AuthRequest
+  | SignRequest
+  | CollectRequest
+  | CancelRequest;
+
+export type BankIdResponse =
+  | CancelResponse
+  | AuthResponse
+  | SignResponse
+  | CollectResponse;
+
+//
+// Client settings
+//
+
+interface BankIdClientSettings {
+  production: boolean;
+  refreshInterval?: number;
+  pfx?: string;
```

Comment:
This can be passed as either a file path (string) or a buffer with the file contents
```suggestion
  pfx?: string | Buffer;
```